src/main.rs

```diff
@@ -67,7 +66,11 @@ fn main() {
             SubCommand::with_name("hook")
                 .about("Runs the shell hook. You shouldn't need to run this manually.")
                 .setting(AppSettings::DisableHelpSubcommand)
-                .arg(Arg::with_name("$__shadowenv_data").required(true))
+                .arg(
+                    // Legacy: This is exported now, and in fact this setting is ignored
+                    // completely if $__shadowenv_data is present in the environment.
+                    Arg::with_name("$__shadowenv_data").required(false)
```

Comment:
I know the deprecation is mentioned in the help, but it might also be worth printing a warning if anyone uses the arg in an unsupported way.